junit5/src/main/kotlin/io/dbkover/junit5/ExtensionContext.kt

```diff
@@ -23,7 +26,25 @@ internal fun ExtensionContext.findDataBaseConnectionFactory(): () -> Connection
             connectionMethod.invoke(requiredTestInstance) as Connection?
                 ?: throw RuntimeException("Connection not initialized correctly")
         }
+    } else if (connectionMethod == null) {
+        return { obtainConnectionFromProperties() ?: throw RuntimeException("Connection not initialized correctly") }
     }
 
     throw RuntimeException("Connection method is missing")
 }
+
+private fun obtainConnectionFromProperties(): Connection? {
+    val url: String? = System.getProperty("datasources.default.url")
+    val username: String? = System.getProperty("datasources.default.username")
+    val password: String? = System.getProperty("datasources.default.password")
```

Comment:
I think that we should add properties to be namespaced with `dbkover` to ensure it doesn't conflict with anything else and not to rely on a specific frameworks structure.

I've added a suggestion below for the property structure 😊

```suggestion
    val url: String? = System.getProperty("dbkover.connection.url")
    val username: String? = System.getProperty("dbkover.connection.username")
    val password: String? = System.getProperty("dbkover.connection.password")
```

Reply:
While I agree we should be framework agnostic, my personal experience is that `datasources.default` is framework agnostic, it is the universally accepted standard to my knowledge. 

What I've seen is that sometimes frameworks also include an optional property that overwrites `datasources.default` but I'm not sure I see the value here.

I'm also a little unsure about the conflict element - wouldn't you always want to run your `dbkover` test against the same default data source as your tests are executed against, if you do not overwrite it with a `@DBKoverConnect` annotated method?

Reply:
I don't believe there is such a standard. Different frameworks has their own way of putting datasources. Here is some examples:

- Spring uses: `spring.datasource` (https://docs.spring.io/spring-boot/appendix/application-properties/index.html#appendix.application-properties.data)
- Quarkus uses: `quarkus.datasource` (https://quarkus.io/guides/datasource)
- Micronaut uses: `datasources` (https://micronaut-projects.github.io/micronaut-sql/latest/guide/configurationreference.html)

Some even use xml configuration for it.

If we should add compatibility with such frameworks (which I think would be nice, but not required), I believe it should be done as separate modules which adds an adapter between DBKover and the given framework 😊

Reply:
Ah, okay! 👍 I was clearly mistaken. I apologise. In that case your suggestion is more than warranted.